Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -692,18 +689,15 @@ function Import-ByDependency
 function Test-AzSOfflineMarketplaceItem {
     param (
         [parameter(mandatory = $true)]
-        [String] $Destination,
-
-        [parameter(mandatory = $true)]
-        [String] $ArmEndpoint,
-
-        [parameter(mandatory = $true)]
-        [String] $SubscriptionId
+        [String] $Destination
     )
 
     $ctx = Get-AzureRmContext
     $AccessToken = Resolve-AccessToken -Context $ctx -AccessToken $AccessToken
     $headers = @{ 'authorization' = "Bearer $AccessToken"}
+    $ArmEndpoint = $ctx.Environment.ResourceManagerUrl
+    $defaultProviderSubscription = Get-AzureRmSubscription -SubscriptionName 'Default Provider Subscription' | Select-AzureRmSubscription
+    $subscriptionId = $defaultProviderSubscription.subscription.id
```

Comment:
This code looks weird and it takes some time to understand what does it do. I would recommend split Get- and Select- statements.

Reply:
fixed, thanks